spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2507,23 +2508,22 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
               partitions.foreach(p => {
                 val inputPartitions = p.asInstanceOf[DataSourceRDDPartition].inputPartitions
                 inputPartitions.foreach(partition => {
-                  partition2Proto(partition.asInstanceOf[FilePartition], nativeScanBuilder)
+                  partition2Proto(partition.asInstanceOf[FilePartition], nativeScanBuilder, scan)
                 })
               })
             case rdd: FileScanRDD =>
               rdd.filePartitions.foreach(partition => {
-                partition2Proto(partition, nativeScanBuilder)
+                partition2Proto(partition, nativeScanBuilder, scan)
               })
             case _ =>
+              assert(false)
           }
 
-          val requiredSchemaParquet =
-            new SparkToParquetSchemaConverter(conf).convert(scan.requiredSchema)
-          val dataSchemaParquet =
-            new SparkToParquetSchemaConverter(conf).convert(scan.relation.dataSchema)
+          val projection_vector: Array[java.lang.Long] = scan.requiredSchema.fields.map(field => {
```

Comment:
This change essentially means that any schema 'adaptation' made in `SparkToParquetSchemaConverter.convert` to support legacy timestamps and decimals will not be supported. But we will probably fail tests with incorrect results.
Also, Comet's Parquet file reader uses `CometParquetReadSupport.clipParquetSchema` to do similar conversion and it includes support for Parquet [field_id](https://github.com/apache/parquet-format/blob/c70281359087dfaee8bd43bed9748675f4aabe11/src/main/thrift/parquet.thrift#L473) which is desirable for delta sources like Iceberg.  
Basically a field_id, if present, identifies a field more precisely (in the event of field name changes) in a schema.

Reply:
The `SchemaConverter` seems like it could be handled in DF's `SchemaAdapter`. I'll look at `clipParquetSchema` as well, thanks!

Reply:
My concern is that...

1. Java side parses Parquet metadata, generates a Spark schema
2. Java side converts Spark schema to Arrow schema (following Comet conversion rules)
3. Serialize Arrow types, native side feeds this into ParquetExec as the data schema

...may yield different results than:

1. Java side serializes original Parquet metadata
2. Serialize schema message
3. Native side parses message, generates Arrow schema and feeds this into ParquetExec as the data schema

I guess I could exhaustively test this hypothesis with all types.